Shopify/cli
# Handle 'unsupported_client_version' error from partners

### WHY are these changes introduced?

Fixes https://github.com/Shopify/develop-app-foundations/issues/270

We want to show a helpful error message for old clients that are trying to deploy.

### WHAT is this pull request doing?

This PR adds error handling in three places:
- GenerateSignedUploadUrl mutation
- UploadUrlGenerate mutation
- CreateDeployment mutation

We can merge this PR whenever we want.

### How to test your changes?

- Try to deploy with partners branch set to `app-deploy-validate-cli-version`.
- You should get this error message

<img width="943" alt="Screenshot 2023-06-07 at 16 25 23" src="https://github.com/Shopify/cli/assets/151725/a050634f-477f-4ae3-bf15-5bf58bde3365">
